akka-javasdk-annotation-processor/src/main/java/akka/javasdk/tooling/processor/ComponentAnnotationProcessor.java

```diff
@@ -93,7 +93,8 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
                   .stream()
                   .collect(Collectors.groupingBy(element -> componentTypeFor(element, annotation)));
                 elementsPerComponentType.forEach((componentType, elements) -> {
-                    var classNames = new ArrayList<>(elements.stream().map(element -> element.getQualifiedName().toString()).toList());
+                    var classNames = new ArrayList<>(elements.stream().map(element ->
+                        processingEnv.getElementUtils().getBinaryName((TypeElement) element).toString()).toList());
```

Comment:
This is the bug fix. getQualifiedName is using `.` as separator, but it should be `$` for our purpose of loading the class